### PR TITLE
fix(Archicad): Timeout after 100s when sending large models #3

### PR DESCRIPTION
**Pull request by jozseflkiss**
## Description & motivation

Solution for https://github.com/specklesystems/speckle-sharp/pull/3048.
Because Grid elements have 3D lines in their 3D representation in Archicad, as a general solution, all kind of edge data is transferred from Archicad to the C# connector. This extra data slowed down the send operation. Since currently we don't story edge information for meshes on the Speckle server, it is not required to transfer all edge data, only the lines without adjacent polygons (which is proper for grids).

## Changes:

C++ connector sends only 3D lines.
Also, time measure points have been introduced for the send operation as well.

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: ConnectorArchicad/ConnectorArchicad/Properties/launchSettings.json
Comment (AlanRynne) [inline ConnectorArchicad/ConnectorArchicad/Properties/launchSettings.json:5]: What are these obscure arguments 😅 Could we get some comments to detail what they mean?

File: ConnectorArchicad/AddOn/Sources/AddOn/Commands/GetModelForElements.cpp
Comment (AlanRynne) [inline ConnectorArchicad/AddOn/Sources/AddOn/Commands/GetModelForElements.cpp:84]: Not sure I fully understand the reasoning behind this, or what it is actually doing.

Could you elaborate more on what other "non 3D lines" were getting sent? And why are they not necessary?

**Review by AlanRynne — CHANGES_REQUESTED** [review]
I've just got 2 questions 👇🏼

File: ConnectorArchicad/ConnectorArchicad/Properties/launchSettings.json
Comment (jozseflkiss) [reply ConnectorArchicad/ConnectorArchicad/Properties/launchSettings.json:5]: The first argument is the HTTP port of the Archicad JSON API, and the second is the Archicad version number (for testing purposes I use very distinct numbers). I'll remove this file from the PR.

File: ConnectorArchicad/AddOn/Sources/AddOn/Commands/GetModelForElements.cpp
Comment (jozseflkiss) [reply ConnectorArchicad/AddOn/Sources/AddOn/Commands/GetModelForElements.cpp:84]: The are vertices, edges and polygons for meshes in Archicad. The edge.pgon1 and edge.pgon2 are the indices of the neighbour polygons of the edge (there are max two of them). If the edge.pgon1 and edge.pgon2 are InvalidPgonIdx, it means that edge is just a line in the 3D model with not polygons attached. This is the case for grid elements. So only these kind of edge/lines are sent to speckle. (Edge status for Mesh elements are not supported yet.)

Comment (AlanRynne) [reply ConnectorArchicad/AddOn/Sources/AddOn/Commands/GetModelForElements.cpp:84]: Oohhh, ok I get it now. This sounds legit 🙌🏼 flagging this as resolved

File: ConnectorArchicad/ConnectorArchicad/Properties/launchSettings.json
Comment (AlanRynne) [reply ConnectorArchicad/ConnectorArchicad/Properties/launchSettings.json:5]: Beautiful, i'll wait for this removal and aprove+merge

**Review by AlanRynne — APPROVED** [review]
Approving after first review + chat with Jozsef. We should release an `rc` to test this in the release version.

**Pull request merged**
